scripts/uberenv/spack_configs/config.yaml

```diff
@@ -112,7 +126,7 @@ config:
   # of Spack.  For filesystems that don't support locking, you should set
   # this to false and run one Spack at a time, but otherwise we recommend
   # enabling locks.
-  locks: false
+  locks: true
```

Comment:
I worry about this one being true, it has caused hangs in the past

Reply:
Lot of change with lock management is Spack since.
I would suggest giving it a chance. I’ve had issues with this _not_ being set to true.

Reply:
It’s is true that this has "only" be tested by me locally and in CI.
So I understand that merging this is less straight-forward than merging CI only.

Reply:
Ok, lets keep them on 
I guess you may actually need it for the CI workflows that have shared spack instances? 

In the past locks weren't necessary and could be a barrier (hangs) for single install. But if we are going to use shared upstreams, we need them.

Reply:
https://github.com/LLNL/conduit/blob/9d654df2a9f93aad23f623a991826d20573ecac9/scripts/ci/build-on-lc.sh#L39

Prefix still depends on the toolchain, I wanted to avoid concurrency at first, but it’s a potential optimization.